audio/src/constant_source.rs

```diff
@@ -0,0 +1,70 @@
+use block::Chunk;
+use block::Tick;
+use node::AudioNodeEngine;
+use node::BlockInfo;
+use node::{AudioNodeType, ChannelInfo};
+use param::{Param, ParamType};
+
+#[derive(Copy, Clone, Debug)]
+pub struct ConstantSourceNodeOptions {
+    pub offset: f32,
+}
+
+impl Default for ConstantSourceNodeOptions {
+    fn default() -> Self {
+        ConstantSourceNodeOptions { offset: 1. }
+    }
+}
+
+#[derive(AudioNodeCommon)]
```

Comment:
According to the [spec](https://webaudio.github.io/web-audio-api/#ConstantSourceNode), `ConstantSourceNode` inherits from `AudioScheduledSourceNode`. So we need to derive `AudioScheduledSourceNode` as well:

```rust
#[derive(AudioScheduledSourceNode, AudioNodeCommon)]
```

Deriving `AudioScheduledSourceNode` adds all these [functions](https://github.com/servo/media/blob/master/servo-media-derive/src/lib.rs#L21) (`should_play_at`, `start`, etc.), so you will also need to add the parameters used in them: `start_at`, `stop_at` and `onended_callback`.

I see that you used `GainNode` as an example to develop this node. I would use `OscillatorNode` instead as a better example, as it inherits from `AudioScheduledSourceNode` as well.